### ISSUE TITLE
FAILED h5netcdf/tests/test_h5netcdf.py::test_group_names

### ISSUE THREAD

**Issue by Hellseher** [h5netcdf/h5netcdf#177]
#### What happened:
While packing this project for Guix I've faced with issue of failing tests and NumPy deprication.

``` shell
=================================== FAILURES ===================================
_______________________________ test_group_names _______________________________

tmp_local_netcdf = '/tmp/guix-build-python-h5netcdf-1.0.1.drv-0/pytest-of-nixbld/pytest-0/test_group_names0/testfile.nc'

    def test_group_names(tmp_local_netcdf):
        # https://github.com/h5netcdf/h5netcdf/issues/68
        with netCDF4.Dataset(tmp_local_netcdf, mode="w") as ds:
            for i in range(10):
                ds = ds.createGroup(f"group{i:02d}")

        with netCDF4.Dataset(tmp_local_netcdf, "r") as ds:
>           assert ds.name == "/"

h5netcdf/tests/test_h5netcdf.py:1623:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   KeyError: 'name'

netCDF4/_netCDF4.pyx:2940: KeyError
=============================== warnings summary ===============================
test_h5netcdf.py: 12 warnings
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:90: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. I
f you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    a, b = map(np.array, (a[...], b[...]))

test_h5netcdf.py::test_write_legacyapi_read_netCDF4
test_h5netcdf.py::test_write_legacyapi_read_netCDF4
test_h5netcdf.py::test_write_legacyapi_read_netCDF4
test_h5netcdf.py::test_write_legacyapi_read_netCDF4
test_h5netcdf.py::test_write_h5netcdf_read_netCDF4
test_h5netcdf.py::test_write_h5netcdf_read_netCDF4
test_h5netcdf.py::test_write_h5netcdf_read_netCDF4
test_h5netcdf.py::test_write_h5netcdf_read_netCDF4
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:90: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    a, b = map(np.array, (a[...], b[...]))

test_h5netcdf.py::test_write_legacyapi_read_netCDF4
test_h5netcdf.py::test_write_h5netcdf_read_netCDF4
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:297: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    assert v.getncattr("_FillValue") == b"X"
 est_h5netcdf.py::test_reopen_file_different_dimension_sizes
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:937: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe.
If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    assert f.groups["one"].variables["foo"][...].shape == (1,)

test_h5netcdf.py::test_expanded_variables_netcdf4[h5netcdf.legacyapi]
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:1505: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe.
 If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.testing.assert_allclose(f.variables["dummy1"][:], res1)

test_h5netcdf.py::test_expanded_variables_netcdf4[h5netcdf.legacyapi]
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:1507: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe.
 If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.testing.assert_allclose(f.variables["dummy2"][:], res2)

test_h5netcdf.py::test_expanded_variables_netcdf4[h5netcdf.legacyapi]
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:1509: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe.
 If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.testing.assert_allclose(f.variables["dummy3"][:], res3)

test_h5netcdf.py::test_expanded_variables_netcdf4[h5netcdf.legacyapi]
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:1511: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe.
 If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.testing.assert_allclose(f.variables["dummy4"][:], res4)

test_h5netcdf.py::test_array_attributes
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:2004: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    assert ds.unicode_fixed == unicode

test_h5netcdf.py::test_array_attributes
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:2005: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    assert ds.unicode_fixed_0dim == unicode

test_h5netcdf.py::test_array_attributes
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:2009: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    assert ds.ascii_fixed == ascii
    
test_h5netcdf.py::test_array_attributes
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:2010: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    assert ds.ascii_fixed_0dim == ascii

test_h5netcdf.py::test_array_attributes
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:2014: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    assert ds.bytes_fixed == ascii

test_h5netcdf.py::test_array_attributes
  /tmp/guix-build-python-h5netcdf-1.0.1.drv-0/h5netcdf-1.0.1/h5netcdf/tests/test_h5netcdf.py:2015: DeprecationWarning: tostring() is deprecated. Use tobytes() instead.
    assert ds.bytes_fixed_0dim == ascii

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED h5netcdf/tests/test_h5netcdf.py::test_group_names - KeyError: 'name'
============ 1 failed, 70 passed, 22 skipped, 33 warnings in 2.20s =============   
```
    
#### What you expected to happen:
All test passed successfully.
#### MCVE Code Sample

Input package versions
``` shel
guix describe
Generation 305  Jun 29 2022 19:31:17    (current)
  guix c92e1b3
    repository URL: https://git.savannah.gnu.org/git/guix.git
    branch: master
    commit: c92e1b3b3170655fc6369a77ad3635fe219a537a

guix show python-setuptools-scm python-pytest python-netcdf4 python-h5py python-packaging python-numpy python | grep -A1 name:
name: python-setuptools-scm
version: 6.3.2
--
name: python-pytest
version: 6.2.5
--
name: python-netcdf4
version: 1.5.3
--
name: python-h5py
version: 3.6.0
--
name: python-packaging
version: 21.3
--
name: python-packaging
version: 21.3
--
name: python-numpy
version: 1.21.6
--
name: python
version: 3.9.9
```
Way to reproduce
```
git clone https://github.com/Hellseher/guix-channel
cd https://github.com/Hellseher/guix-channel
git checkout a43c3ebd55cd2cd00302463978ea825505c54343
guix build -K -L . python-h5netcdf
```

#### Version
- https://github.com/h5netcdf/h5netcdf/tree/v1.0.1

**Comment by kmuehlbauer** [comment]
@Hellseher Thanks for the report. It's good to see people actually start using `h5netcdf` > 1.0. Quick question: which version of netcdf-c and hdf5 is netCDF4 using in your system?

**Comment by kmuehlbauer** [comment]
@Hellseher It turns out that version 1.5.3 of `netCDF4` is missing `.name` which was reported here:  https://github.com/Unidata/netcdf4-python/issues/988. This was fixed in `netCDF4` version 1.5.4. So updating to at least that version will fix the failing test.

At the same time this should also fix the DeprecationWarnings, as this has been fixed in the same version, AFAICT.

**Comment by Hellseher** [comment]
Hi,
Thank you for a quick response!
I've listed input packages which were require for the build in the "MCVE Code Sample"

```
name: python-setuptools-scm
version: 6.3.2
--
name: python-pytest
version: 6.2.5
--
name: python-netcdf4
version: 1.5.3
--
name: python-h5py
version: 3.6.0
--
name: python-packaging
version: 21.3
--
name: python-packaging
version: 21.3
--
name: python-numpy
version: 1.21.6
--
name: python
version: 3.9.9
```

Your projects goes as additional dependencies in SunPy [h5netcdf>=0.8.1](https://github.com/sunpy/sunpy/blob/ec1135631c5ab953729416d0560286ab403684a8/setup.cfg#L70) But I've tried to pack the lates version.

**Comment by kmuehlbauer** [comment]
The problem is not with `h5netcdf=1.0.1` but with `python-netcdf4==1.5.3` it just surfaces in our tests. You would need to upgrade `python-netcdf4`. That will fix all errors and warnings you are currently seeing.

**Comment by Hellseher** [comment]
Just to confirm and close this issue:
I've bumped `python-netcdf4` to the latest  available in upstream (1.6.0), now all test passed successfully!
```
======================== 71 passed, 22 skipped in 2.17s ========================
phase `check' succeeded after 2.7 seconds
```
Thanks again for a quick and positive response :1st_place_medal: 

This issue could be closed.

**Comment by kmuehlbauer** [comment]
@Hellseher You're welcome. Glad it works and the issue could be resolved within one hour :tada:.
